invenio_userprofiles/forms.py

```diff
@@ -68,7 +68,7 @@ def validate_username(form, field):
             user_profile = UserProfile.get_by_username(field.data)
             if current_userprofile.is_anonymous or \
                     (current_userprofile.user_id != user_profile.user_id and
-                     field.data != current_userprofile.username):
+                     field.data == current_userprofile.username):
```

Comment:
First of all, I would split this if, is a messy one, and I agree with the change. But I would do:
```suggestion
if current_userprofile.is_anonymous:
    # it should never happen, that's why we maybe show a different message 
    raise ValidationError(_('Cannot change username for current user.'))

is_same_user = current_userprofile.user_id == user_profile.user_id
if not is_same_user:
    # it should never happend
    raise ValidationError(_('The username cannot be changed.'))

is_same_username = field.data == current_userprofile.username
if is_same_username:
    # do we really have to raise in this case? can't we just skip the change? otherwise:
    raise ValidationError(_('The new username is the same as the previous one. Please provide a different one.'))
```

Reply:
I agree on that this ``if`` needs to be split up - almost impossible to decode what's going and it masks the different cases.

The complexity comes from:
- The profile form is used for both 1) sign up and 2) profile page. 
- Usernames are case-insensitive but stored in a case-preserving way (i.e. you cannot have two users with the same username - i.e. one with ``LNIELSEN`` and the other with ``lnielsen`` - but a single user can freely change between the two.

So I think the validation logic is simpler:

- Sign up: If a username already exists, FAIL
- Profile form: If a username already exists, and it's not the same user (i.e. id's mismatch). FAIL 

No need to check the username.